helper/helper.go

```diff
@@ -363,3 +366,27 @@ func TLScertToFile(filename string, derBytes []byte) error {
 
 	return err
 }
+
+// FormatUploadFilePath ensures that path separators are "/", and returns error if the
+// filepath contains a disallowed character matched with regex
+func FormatUploadFilePath(filePath string) (string, error) {
+
+	// Check for mixed "\" and "/" in filepath. Stop and throw an error if true so that
+	// we do not end up with unintended folder structure when applying ReplaceAll below
+	if strings.Contains(filePath, "\\") && strings.Contains(filePath, "/") {
+		return filePath, fmt.Errorf("filepath contains mixed '\\' and '/' characters")
+	}
+
+	// make any windows path separators linux compatible
+	outPath := strings.ReplaceAll(filePath, "\\", "/")
```

Comment:
yes, but that one only converts `os.PathSeparator` to `/`
Here we have a specific case with escaped backslashes that needs to be converted.

Reply:
Yeah I thought so too but, it turns out `ToSlash` is not a good choice here since we have the case where `proxy` always runs in linux but s3clients that make incoming requests run on a host OS that is unknown to `proxy`.